File: Source/Aladdin.CAPI.GOST.Java/src/aladdin/asn1/gost/GOST28147CipherParameters.java
package aladdin.asn1.gost;
import aladdin.asn1.*; 
import java.io.*; 

//	GOST28147Parameters ::= SEQUENCE {
//		iv                   OCTET STRING (SIZE (8)),
//		encryptionParamSet   OBJECT IDENTIFIER
//	}

public final class GOST28147CipherParameters extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(OctetString		.class).factory(8, 8), Cast.N), 
		new ObjectInfo(new ObjectCreator(ObjectIdentifier	.class).factory(    ), Cast.N), 
	}; 
	// конструктор при раскодировании
	public GOST28147CipherParameters(IEncodable encodable) throws IOException { super(encodable, info); }
    
	// конструктор при закодировании
	public GOST28147CipherParameters(OctetString iv, ObjectIdentifier paramSet)
	{
		// проверить ограничение
		super(info, iv, paramSet); 
	}  
	public final OctetString		iv		()	{ return (OctetString	  )get(0); } 
	public final ObjectIdentifier	paramSet()	{ return (ObjectIdentifier)get(1); }
}

File: Source/Aladdin.CAPI.GOST/ASN1/GOST/GOST28147CipherParameters.cs
using System;
using System.Runtime.Serialization; 

//	GOST28147Parameters ::= SEQUENCE {
//		iv                   OCTET STRING (SIZE (8)),
//		encryptionParamSet   OBJECT IDENTIFIER
//	}

namespace Aladdin.ASN1.GOST
{
    [Serializable]
	public class GOST28147CipherParameters : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<OctetString		>().Factory(8, 8), Cast.N), 
			new ObjectInfo(new ObjectCreator<ObjectIdentifier	>().Factory(    ), Cast.N), 
		}; 
		// конструктор при сериализации
        protected GOST28147CipherParameters(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public GOST28147CipherParameters(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public GOST28147CipherParameters(OctetString iv, ObjectIdentifier paramSet) : 
			base(info, iv, paramSet) {}  

		public OctetString		IV			{ get { return (OctetString		)this[0]; } } 
		public ObjectIdentifier	ParamSet	{ get { return (ObjectIdentifier)this[1]; } }
	}
}
